core/finality-grandpa/src/lib.rs

```diff
@@ -136,6 +137,11 @@ const LAST_COMPLETED_KEY: &[u8] = b"grandpa_completed_round";
 const AUTHORITY_SET_KEY: &[u8] = b"grandpa_voters";
 const CONSENSUS_CHANGES_KEY: &[u8] = b"grandpa_consensus_changes";
 
+const GRANDPA_ROUND_MESSAGE: u32 = 0x1001;
+const GRANDPA_COMMIT_MESSAGE: u32 = 0x1002;
```

Comment:
these constants seems like they could be easily conflicting with other modules

Reply:
They are part of the network message effectively. Any suggestions here?

Reply:
Making sure these `const`s don't conflict (especially when people bring in their own consensus algorithms) sounds like a nightmare :( ...

Reply:
@arkpar as part of #1304 I've proposed that we make digest items and other substrate primitives start with a 4-byte "magic number" which are assigned to consensus engines sort of like port numbers. We'll reserve some numbers and an early range for some consensus engines and custom implementors should do the same.

In light of that we could change the gossip packet structure to

`{ consensus_engine: [u8; 4], message_kind: u8, data: Vec<u8> }` instead.

Reply:
Alternatively we could group gossip validators by `consensus_engine` as opposed to message kind, and have the message kind be part of an encoded `enum` in `grandpa::communication`

Reply:
Implemented the second option